src/Xamarin.Android.Build.Tasks/Utilities/ApplicationConfig.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+
+namespace Xamarin.Android.Tasks
+{
+	// Order of fields and their types must correspond *exactly* to that in
```

Comment:
May need to elaborate on the meaning of "exactly".  "Exactly" at the ABI level?  Other?

I'm not even sure what C++ `bool` *is* at the ABI level (i.e. what is `sizeof(bool)` in C++? I don't know offhand).  The header uses `uint32_t` while C# uses `uint`, which is a 32-bit int, so these are the same in terms of ABI, but only "semantically the same" at the API level.

Then there's `string android_package_name`, and I don't know what brain bending I should use to associate C# `string` with C++ `const char *`.

("ABI -ish"?)

Reply:
This is all about declaration order of fields and their bit size. I will add a comment with the mapping.